### PR TITLE
Units tests

### PR DESCRIPTION
**Pull request by GhostEsso**
Hello **Code Reviewer 👋🏾**,
We hope you are doing well 🤓!!!

In this Milestone, We  implemented the following on this branch:

unit tests for :
- Book, 
- Person, 
- Student, 
- Teacher, 
- Classroom, 
- Rental,
-  All classes implemented for the Decorator design pattern

Thanks :pray:  for reviewing our Pull Request, **stay blessed :star2:**

### CODE REVIEW

File: spec/classroom_spec.rb
Comment (shahadat3669) [inline spec/classroom_spec.rb:1]: 👍 Good job on adding a test case for the `classroom` class instance! However, it's important to cover all the methods with test cases as per the project requirements. Currently, you haven't added test cases for the `add_student` method. Please consider adding test cases for all methods in your files to ensure comprehensive test coverage. Keep up the great work! 👏🧪📋

File: spec/person_spec.rb
Comment (shahadat3669) [inline spec/person_spec.rb:1]: - [ ] 👍 Good job on adding a test case for the `classroom` class instance and for the `can_use_services` method! However, it's important to cover all the methods with test cases as per the project requirements. Currently, you haven't added test cases for the `correct_name`, `classroom`, and `add_new_rental` methods. Please consider adding test cases for all methods in your files to ensure comprehensive test coverage. Keep up the great work! 👏🧪📋

File: spec/student_spec.rb
Comment (shahadat3669) [inline spec/student_spec.rb:1]: - [ ] 👍 Good job on adding a test case for the `classroom` class instance and for the `play_hooky` method! However, it's important to cover all the methods with test cases as per the project requirements. Currently, you haven't added test cases for the `classroom` method. Please consider adding test cases for all methods in your files to ensure comprehensive test coverage. Keep up the great work! 👏🧪📋

File: spec/teacher_spec.rb
Comment (shahadat3669) [inline spec/teacher_spec.rb:1]: - [ ] 👍 Good job on adding a test case for the `classroom` class instance! However, it's important to cover all the methods with test cases as per the project requirements. Currently, you haven't added test cases for the `can_use_services` method. Please consider adding test cases for all methods in your files to ensure comprehensive test coverage. Keep up the great work! 👏🧪📋

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @GhostEsso, @mohammadnewton, 

Great job so far! 👏 You've done well, and you're almost there for the next project ⏭️, but there are a few issues 🐛 that you still need to work on.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ All current test cases pass
✅ All test files are in a different directory (e.g /spec)
✅ there is an 'it' block for each class method 💯


## Required Changes ♻️
- [ ] 👍 Great job on thoroughly testing most of your project files! That's commendable. However, to align with the project requirements, it's crucial to ensure that all classes you've created while implementing the Decorator design pattern have test cases. Currently, I couldn't find test files for the `decorator` and `nameable` classes. Kindly consider adding test cases for these classes and updating your code to ensure complete test coverage. Your testing efforts are much appreciated! 👏🧪📦

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by divinecharlotte — APPROVED** [review]
# Hello @GhostEsso  😃 🥳 🎉 🎈 

Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say other than... it's time to merge it 👍🏻 ✅ 
### To Highlight 💯

- Good implementation of the required changes✔️
- Descriptive pr✔️






## Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**